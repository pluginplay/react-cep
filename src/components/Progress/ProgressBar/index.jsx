import React from 'react'
import ProgressContainer from '../ProgressContainer'
import ProgressInner from '../ProgressInner'
import PropTypes from 'prop-types'

class ProgressBar extends React.Component {

  render () {
    return (
      <ProgressContainer>
        <ProgressInner percentage={this.props.percentage} isComplete={this.props.isComplete} />
      </ProgressContainer>
    )
  }

}

ProgressBar.propTypes = {
  percentage: PropTypes.number.isRequired,
  isComplete: PropTypes.bool.isRequired
}

export default ProgressBar
