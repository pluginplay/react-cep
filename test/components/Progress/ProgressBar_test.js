import React from 'react'
import ProgressBar from '../../../src/components/Progress/ProgressBar'
import renderer from 'react-test-renderer'
import { withTheme } from '../../setup'

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<ProgressBar percentage={20} isComplete={false} />))
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when complete', () => {
  const tree = renderer
    .create(withTheme(<ProgressBar percentage={20} isComplete={true} />))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
