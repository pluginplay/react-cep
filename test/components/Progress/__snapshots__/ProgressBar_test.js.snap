// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa dDyNSC"
>
  <div
    className="sc-bwzfXH hcbpUg"
  />
</div>
`;

exports[`renders correctly when complete 1`] = `
<div
  className="sc-bdVaJa dDyNSC"
>
  <div
    className="sc-bwzfXH hsnZWV"
  />
</div>
`;
